docs/ECOS.md

```diff
@@ -0,0 +1,141 @@
+
+## ECO Runtime Specification
+
+ECO Runtime Specification aims to specify the binding of ECI to Edge Node's hardware and software resources, configuration, execution environment, and lifecycle of an Edge Container. Note that unlike most of the runtimes covered by OCI specification, EVE is a self-contained execution engine. This allows us to skip a lot of details (around on-disk container runtime formats, configuration schema, etc.) that otherwise would have to be standardized.
+
+Since there is always a 1-many mapping between ECI and ECOs, a very important question is how do you reference a seed ECI for any ECO operation requiring it (like creating an ECO from an ECI). There are two methods that are covered in the following section: you can reference an immutable ECI via a content-addressable handle or you can reference an ECI via a human readable name that maps to an immutable one. Since the later kind of mapping can be updated at will, it is very important to understand how various transitions in the ECO state transition diagram react to an ECI mapping changing while transition is in flight.
+
+Since nothing happens with EVE unless initiated by a Controller through a certain set of configuration directives, this section focuses on:
+
+   #1 ECO state transition diagram
+   #2 configuration directives utilized by an EVE controller to move through the states
+   #3 metrics and events communicated back by EVE to the controller after any transition is attempted
+
+### ECO Runtime State Transition Diagram
+
+We will try to align with [OCI Lifecycle](https://github.com/opencontainers/runtime-spec/blob/master/runtime.md#lifecycle) where possible, but we may need some extra states as well.
+
+### Detailed description of states
+
+Currently we support the following states for the ECOs:
+
+* ECI has been transferred to EVE's storage area
+* Resources (CPU, memory, networking, I/O) have been committed to an ECO but the container has not been started yet
+* ECO is running
+* ECO is stopped (the resources are still pinned and mutable resources such as R/W storage remain in the state they were when ECO was running)
+* ECO is restarted
+* ECO resources are purged (TBD)
+* ECO is deleted (all the resources are released)
+
+### Controller config ECO schema
+
+As part of a regular config that EVE recieves from its controller, each ECO gets configured by an entry in an array having:
+
+* displayname
+* uuidandversion
+* activate
+* purge
+* restart
+* fixedresources
+  * virtualizationMode
+    * memory
+    * maxmem
+    * vcpus
+    * rootdev
+  * interfaces
+    * name
+    * acls
+    * networkId
+  * drives
+    * drvtype
+    * target
+    * maxsizebytes
+    * readonly
+    * preserve
+    * image
+      * name
+      * uuidandversion
+        * iformat
+        * sizeBytes
+        * sha256
+        * dsId
+        * siginfo
+
+## ECO Image Format
```

Comment:
Should this title not be "Edge Container Image Format" ("ECO" implies _not_ "ECI")

Reply:
Good catch. Fixed.